pkg/controllers/management/drivers/nodedriver/machine_driver.go

```diff
@@ -84,24 +84,27 @@ type Lifecycle struct {
 	dockerMachineVersion string
 }
 
-func (m *Lifecycle) Create(obj *v3.NodeDriver) (runtime.Object, error) {
+func (m *Lifecycle) Create(obj *v32.NodeDriver) (runtime.Object, error) {
 	return m.download(obj)
 }
 
-func (m *Lifecycle) download(obj *v3.NodeDriver) (*v3.NodeDriver, error) {
+func (m *Lifecycle) download(obj *v32.NodeDriver) (*v32.NodeDriver, error) {
 	driverLock.Lock()
 	defer driverLock.Unlock()
 	if !obj.Spec.Active && !obj.Spec.AddCloudCredential {
 		return obj, nil
 	}
 
 	forceUpdate := m.checkDriverVersion(obj)
+	if v32.NodeDriverConditionDownloaded.GetStatus(obj) == "" || v32.NodeDriverConditionInstalled.GetStatus(obj) == "" {
```

Comment:
I added this in the event the driver was downloaded, but the status wasn't updated, which is really to deal with edge cases.